functions/index.js

```diff
@@ -0,0 +1,104 @@
+const functions = require('firebase-functions');
+
+// gMail API constants.
+const fs = require('fs');
+const readline = require('readline');
+const {google} = require('googleapis');
+const TOKEN_PATH = 'token.json';
+
+// The file token.json stores the user's access and refresh tokens, and is
+// created automatically when the authorization flow completes for the first
+// time.
+// If modifying the scopes below, delete token.json.
+const SCOPES = [
+  'https://mail.google.com/',
+  'https://www.googleapis.com/auth/gmail.modify',
+  'https://www.googleapis.com/auth/gmail.compose',
+  'https://www.googleapis.com/auth/gmail.send',
+];
+
+exports.sendMail = functions.https.onRequest((request, response) => {
```

Comment:
This line should be changed in the future -- from httpsRequest to firestore listener.

Reply:
Done.